core/src/kura.rs

```diff
@@ -463,15 +505,39 @@ async fn storage_files_base_indices<IO: DiskIO>(
     path: &Path,
     io: &IO,
 ) -> Result<BTreeSet<NonZeroU64>> {
-    let bases = io
+    let dirs: Vec<(String, IO::Dir)> = io
         .read_dir(path.to_path_buf())
         .await?
         .filter_map(|item| async {
```

Comment:
Don't you think it looks prettier? I'm talking about using `filter` and `map` instead of `filter_map`.

AFAIK we should use `filter_map` when we will apply an operation to parameter regardless of its value. Here is not this situation, here we apply operation only on `Ok` variant.